src/web/app/pages-instructor/instructor-courses-page/add-course-form/add-course-form.component.ts

```diff
@@ -131,7 +132,9 @@ export class AddCourseFormComponent implements OnInit {
           this.statusMessageService.showErrorToast(resp.error.message);
         });
 
-        modalRef.result.then((result: CopyCourseModalResult) => this.copyCourseEvent.emit(result), () => {});
+        await modalRef.result
```

Comment:
Hmm can I clarify which function call inside is async? `showErrorToast`?

Reply:
I am not very sure but my understanding of `await` is that it is used on `Promise` objects and in this case `modalRef.result` is a Promise. 
```
        try {
          await modalRef.result
          const result: CopyCourseModalResult = await modalRef.result;
          this.copyCourseEvent.emit(result)
        } catch (resp: any) {
          this.statusMessageService.showErrorToast(resp.error.message);
        }
```
Would it be correct to change this part to the above?

Reply:
As in: is there any reason to make this a blocking call?
`subscribe` itself is already a Promise-ish thing. I don't see why you need an await here even tho `modalRef.result` is a Promise.

Reply:
Ok I see, then I think the async and await is not needed here.